src/components/molecules/error-component.tsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import {View, StyleSheet} from 'react-native';
+import {TextView, CustomButton} from '../atoms';
+import label from '../../translations/en.json';
+
+interface ErrorProps {
+  message: string;
+  buttonText?: string;
+  onButtonPress?: () => void;
+}
+
+const ErrorComponent: React.FC<ErrorProps> = ({
+  message,
+  buttonText = label.try_again,
+  onButtonPress,
+}) => {
+  return (
+    <View style={styles.container}>
+      <TextView style={styles.message}>{message}</TextView>
+      {onButtonPress && (
```

Comment:
I'm confused. When an error occurs, there is currently no option for the user to go back; user can only retry. I think it would be nice to have a back option.

Reply:
@Asad-wego  I've implemented back button option into the error component, allowing users to navigate back in case of errors,as per your suggestion. Thank you!